kyaml/yaml/rnode_test.go

```diff
@@ -721,6 +721,288 @@ data:
+`,
+			expected: `
+apiVersion: v1
+kind: MergeTagTest
+metadata:
+  name: test
+data:
+  color:
+    foo: bar
+  primaryColor:
+    foo: bar
+`,
+		},
+		// *********
+		// Test Case
+		// *********
+		{
+			description: "keep duplicated keys",
+			input: `
+apiVersion: v1
+kind: MergeTagTest
+metadata:
+  name: test
+data:
+  color: "#FF0000"
+  color: "#FF00FF"
+`,
+			expected: `
+apiVersion: v1
+kind: MergeTagTest
+metadata:
+  name: test
+data:
+  color: "#FF0000"
+  color: "#FF00FF"
+`,
+		},
+		// *********
+		// Test Case
+		// *********
+		{
+			description: "keep json",
+			input:       `{"apiVersion": "v1", "kind": "MergeTagTest", "spec": {"color": {"rgb": "#FF0000"}}}`,
+			expected:    `{"apiVersion": "v1", "kind": "MergeTagTest", "spec": {"color": {"rgb": "#FF0000"}}}`,
+		},
+		// *********
+		// Test Case
+		// *********
+		{
+			description: "keep comments",
+			input: `
+apiVersion: v1
+kind: MergeTagTest
+metadata:
+  name: test
+data:
+  color: &color-used
+    foo: bar
+  primaryColor:
+    # use same color because is pretty
+    rgb: "#FF0000"
+`,
+			expected: `
+apiVersion: v1
+kind: MergeTagTest
+metadata:
+  name: test
+data:
+  color:
+    foo: bar
+  primaryColor:
+    # use same color because is pretty
+    rgb: "#FF0000"
+`,
+		},
+		// *********
+		// Test Case
+		// *********
+		{
+			description: "works with explicit merge tag",
+			input: `
+apiVersion: v1
+kind: MergeTagTest
+metadata:
+  name: test
+data:
+  color: &color-used
+    foo: bar
+  primaryColor:
+    !!merge <<: *color-used
+`,
+			expected: `
+apiVersion: v1
+kind: MergeTagTest
+metadata:
+  name: test
+data:
+  color:
+    foo: bar
+  primaryColor:
+    foo: bar
+`,
+		},
+		// *********
+		// Test Case
+		// *********
+		{
+			description: "merging properties",
+			input: `
+apiVersion: v1
+kind: MergeTagTest
+metadata:
+  name: test
+data:
+  color: &color-used
+    rgb: "#FF0000"
+  primaryColor:
+    <<: *color-used
+    pretty: true
+`,
+			expected: `
+apiVersion: v1
+kind: MergeTagTest
+metadata:
+  name: test
+data:
+  color:
+    rgb: "#FF0000"
+  primaryColor:
+    pretty: true
+    rgb: "#FF0000"
+`,
+		},
+		// *********
+		// Test Case
+		// *********
+		{
+			description: "overriding value",
+			input: `
+apiVersion: v1
+kind: MergeTagTest
+metadata:
+  name: test
+data:
+  color: &color-used
+    rgb: "#FF0000"
+    pretty: false
+  primaryColor:
+    <<: *color-used
+    pretty: true
+`,
+			expected: `
+apiVersion: v1
+kind: MergeTagTest
+metadata:
+  name: test
+data:
+  color:
+    rgb: "#FF0000"
+    pretty: false
+  primaryColor:
+    pretty: true
+    rgb: "#FF0000"
+`,
+		},
+		// *********
+		// Test Case
+		// *********
+		{
+			description: "merging multiple anchors",
+			input: `
+apiVersion: v1
+kind: MergeTagTest
+metadata:
+  name: test
+data:
+  color: &color
+    rgb: "#FF0000"
+    pretty: false
+  primaryColor: &primary
+    rgb: "#0000FF"
+    alpha: 50
+  secondaryColor:
+    <<: *color
+    <<: *primary
```

Comment:
go-yaml also doesn't consider this valid, because technically it is a duplicate key:

> line 15: mapping key "<<" already defined at line 14

According to the spec, if you want to merge multiple anchors, the correct syntax is `<< : [ *color, *primary ]`. I'm not sure whether a 1.1 implementation would allow the above, but since we're (via go-yaml) a 1.2 implementation, I think we shouldn't support it at this level either.

Reply:
I was testing things out with [yq](https://github.com/mikefarah/yq), which uses `go-yaml`, and it worked fine with this duplicate key.
The new implementation makes it work with both. We can change that, but for me, it seems that this type of check would be in another part of the code... that's why I left any duplicate keys as is

Reply:
I suppose for me it depends on how difficult it is. I agree with @KnVerey that ideally we would not support it, and it would be better to throw an error instead. However based on @RafaeLeal's comment:

> We can change that, but for me, it seems that this type of check would be in another part of the code...

Maybe this check is out of scope for this PR? I would say that if it is not terribly difficult to do the check then let's do it, otherwise maybe it is not worth the effort - though I would hate for it to become a regression in the future. 

I recall a portion of code that throws an error in the case of duplicate keys. @RafaeLeal if you run this case end to end with `kustomize build`, does kustomize itself throw an error somewhere?

Reply:
Chatted with @KnVerey and it looks like duplicate key detection here will be easy to catch. I agree that we should throw an error in this test case.

Reply:
Now it's returning an error `duplicated merge key`